src/utils/get-server-status.ts

```diff
@@ -0,0 +1,181 @@
+import { Config } from './../interfaces/config'
+import { ServerStatus, Status } from './../interfaces/server-status'
+import { Probe } from '../interfaces/probe'
+import { ValidateResponseStatus } from './alert'
+import console, { log } from 'console'
+
+const SERVER_STATUSES: ServerStatus[] = []
+
+// Function to calculate successes and failures, either total or consecutives.
+// If consecutive success/failure equals to threshold, it will send notifications
+const calculateSuccessFailure = ({
+  errorName,
+  probeStatus,
+  validation,
+  threshold,
+}: {
+  errorName: string
+  probeStatus: Status
+  validation: ValidateResponseStatus
+  threshold: number
+}) => {
+  log(`Checking for ${errorName} in probe...`)
+  switch (probeStatus.isDown) {
+    case false:
+      // If current probe is UP and trigger equals true
+      if (validation.status) {
+        // Add failure count
+        probeStatus.shouldSendNotification = false
```

Comment:
Can we wrap/encapsulate, in separate function, for every state (combination of `isDown`, `validation.status` and `consecutiveSuccesses === threshold`)?

Reply:
Sure